data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -301,17 +319,28 @@ public Map<String, String> getTracingAttributes() {
         return attributes;
     }
 
+    /**
+     * Gets the amount of time a query may execute on the service before it times out, formatted as a KQL timespan.
+     * @param timeoutObj amount of time before timeout, which may be a Long, String or Integer.
+     *                    Value must be between 1 minute and 1 hour, and so value below the minimum or above the maximum will be adjusted accordingly.
+     */
+    // TODO: Breaking change for next major release: static String getTimeoutAsTimespan(Object timeoutObj)
     String getTimeoutAsString(Object timeoutObj) {
-        String timeoutString = "";
-        if (timeoutObj instanceof Long) {
-            Duration duration = Duration.ofMillis((Long) timeoutObj);
-            timeoutString = Utils.formatDurationAsTimespan(duration);
-        } else if (timeoutObj instanceof String) {
-            timeoutString = (String) timeoutObj;
-        } else if (timeoutObj instanceof Integer) {
-            Duration duration = Duration.ofMillis((Integer) timeoutObj);
-            timeoutString = Utils.formatDurationAsTimespan(duration);
+        Long timeoutInMilliSec = getTimeoutInMilliSec(timeoutObj);
+
+        if (timeoutInMilliSec == null) {
+            return null;
         }
-        return timeoutString;
+
+        Duration duration = Duration.ofMillis(timeoutInMilliSec);
+        return Utils.formatDurationAsTimespan(duration);
+    }
+
+    /**
+     * Gets the amount of time a query may execute on the service before it times out, formatted as a KQL timespan.
+     * Value must be between 1 minute and 1 hour, and so if the value had been set below the minimum or above the maximum, the value returned will be adjusted accordingly.
+     */
+    String getTimeoutAsTimespan() {
```

Comment:
It makes more sense to have this method than the existing `String getTimeoutAsString(Object timeoutObj)`.

Also, it's consistent with the existing `public Long getTimeoutInMilliSec()`.

I kept the access modifier consistent with the existing method. I suggest changing this to `public` when we make the breaking change I suggest in the previous method, since the different namings (`getTimeoutAsString` vs `getTimeoutAsTimespan`) may be confusing. Or I can do that now.

Reply:
I feel like with this name it's expected to return a Timespan object of some sorts. Maybe a name like getTimeoutFormatted would show the intention better? can you give other names that are consistent with that?

Reply:
I see the naming concern here. OTOH, I think it can also intuitively makes sense that an "As/In" doesn't necessarily mean the object's literal `class`, but instead the business description.

For example, `getTimeoutInMilliSec()` returns a Long, which I think is also intuitive.

Reply:
I don't think we use the Csl terminology much.

how about formatTimeoutAsTimespan ?

Reply:
That's what we call it, including in the Java SDK:
`public class CslTimespanFormat extends CslFormat`

Reply:
I guess `getTimeoutFormattedAsTimespan()` could work, but that sounds overly verbose to me.